terraform/aws/modules/eks/eks.tf

```diff
@@ -0,0 +1,39 @@
+resource "aws_iam_role" "demo" {
+  name = "eks-cluster-demo"
```

Comment:
Please dont use names like demo. These will be hardcoded in the state. I'd instead suggest default, or just eks_role, etc. Please update this everywhere you've used demo.